hadoop-ozone/dev-support/checks/findbugs.sh

```diff
@@ -16,16 +16,22 @@
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
 cd "$DIR/../../.." || exit 1
 
-mvn -B compile -fn findbugs:check -Dfindbugs.failOnError=false  -f pom.ozone.xml
+if ! type unionBugs >/dev/null 2>&1 || ! type convertXmlToText >/dev/null 2>&1; then
+  mvn -B -fae compile spotbugs:check -f pom.ozone.xml
```

Comment:
Can you please help me to understand this part? Why don't we need mvn execution if both unionBugs and convertXmlToText is on the path?

Reply:
Maven is also run (with slightly different target) if those scripts are available:

https://github.com/apache/hadoop/pull/1513/files/df5ed66f02780284a45e6b0496af8999f68d5bc3#diff-1eb4e94fe21f1565058f335351da12b4R24

The difference:
 * `spotbugs:spotbugs` always succeeds, output is post-processed, and script exit is determined based on that (regular behavior)
 * `spotbugs:check` exits with success/failure status, which is used directly for the script exit value (fallback behavior, in case we don't have the utilities)

Hope that helps.

Reply:
Sure. Thanks. I missed that with the exit it's more like an if/else.

Reply:
Exactly, but I wanted to avoid reindenting the whole remaining script.